nixos/modules/services/networking/lokinet.nix

```diff
@@ -0,0 +1,219 @@
+    };
+
+    nickname = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "lokinet";
+      description = "Sets router nickname to what you want.";
+    };
+
+    minRouters = mkOption {
+      type = types.int;
+      default = 4;
+      example = 6;
+      description = "Maintain minimum connections to other routers.";
+    };
+
+    maxRouters = mkOption {
+      type = types.int;
+      default = 6;
+      example = 8;
+      description = "Hard limit of routers globally we are connected to at any given time.";
+    };
+
+    loggingType = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "syslog";
+      description = "Logging method to use.";
+    };
+
+    loggingFile = mkOption {
+      type = types.path;
+      default = "/var/lib/lokinet/log";
+      example = "/home/okina/lokinet/log";
+      description = "Path to the lokinet log, only enabled if loggingType is set to \"file\".";
+    };
+
+    resolverUpstream = mkOption {
+      type = types.str;
+      default = "1.1.1.1";
+      example = "8.8.8.8";
+      description = "DNS resolver upstream address.";
+    };
+
+    resolverBind = mkOption {
+      type = types.str;
+      default = "127.3.2.1:53";
+      example = "127.3.2.1:52";
+      description = "DNS resolver bind address.";
+    };
+
+    strictConnect = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Add router with pubkey to list of routers we connect directly to.";
+    };
+
+    exitNode = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Use router with pubkey as an exit node";
+    };
+
+    snapp = mkOption {
+      default = null;
+      description = "Snapp configuration.";
+
+      type = types.nullOr (types.submodule {
+        options = {
+          ifAddr = mkOption {
+            type = types.str;
+            default = "10.33.0.1/16";
+            example = "10.34.0.1/16";
+            description = "The ip range to allocate to this snapp.";
+          };
+
+          ifName = mkOption {
+            type = types.str;
+            default = "snapp-tun0";
+            example = "snapp-tun1";
+            description = "The name to try and give to the network interface this snapp owns.";
+          };
+        };
+      });
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd = {
+      services.lokinet = {
+        description = "lokinet";
+        after = [ "network-online.target" ];
+        wantedBy = [ "multi-user.target" ];
+        preStart = "${pkgs.curl}/bin/curl https://seed.lokinet.org/bootstrap.signed -o ${dataDir}/bootstrap.signed";
+
+        serviceConfig = {
+          User = "lokinet";
+          DynamicUser = true;
+          StateDirectory = "lokinet";
+          AmbientCapabilities = [ "cap_net_admin" "cap_net_bind_service=+eip" ];
+          ExecStart = "${pkgs.lokinet}/bin/lokinet";
+        };
+      };
+
+      tmpfiles.rules = [
```

Comment:
As mentioned above you will no longer require `tmpfiles` when you apply the above suggested code.